core/api/src/mill/api/internal/EvaluatorApi.scala

```diff
@@ -7,6 +7,7 @@ trait EvaluatorApi extends AutoCloseable {
   def evaluate(
       scriptArgs: Seq[String],
       selectMode: SelectMode,
+      reporter: Int => Option[CompileProblemReporter] = _ => None,
```

Comment:
I wonder if we should make this a constructor param on `Evaluator` instead? We can e.g. mark every evaluator going into BspEvaluators with `.withCompileProblemReporter`, which would simplify the user-facing `Evaluator` API and ensure BSP code doesn't forget to pass in `reporter` at any use sites

Reply:
the refactor does preserve the public interface of `Evaluator` and prevent it from becoming unnecessarily complicated, so it's actually user facing and not just of interest internally

Reply:
Makes sense.

Reply:
Even if we pass the reporter builder directly to `Evaluator`, it seems we'll still need to pass the "origin id", that can differ for each BSP request, to individual methods. The origin id has to be added to the notifications that the compile problem reporter reports.

Reply:
Ok!